src/utils/formatError.ts

```diff
@@ -0,0 +1,24 @@
+import axios from 'axios';
+import { APIError } from '../api';
+import { IMXError } from '../types/errors';
+
+/**
+ * [Formats an error in the IMXError shape](https://axios-http.com/docs/handling_errors)
+ * @param error - The Error object thrown by the request
+ * @returns IMXError
+ */
+
+export function formatError(error: unknown): IMXError {
+  if (axios.isAxiosError(error) && error.response) {
+    const apiError: APIError = error.response.data;
+    return new IMXError({
+      code: apiError.code,
+      details: apiError.details,
+      message: apiError.message,
+    });
+  }
+  return new IMXError({
+    code: 'unknown_error_code',
```

Comment:
Would it be worth adding this to our docs, [here?](https://docs.x.immutable.com/docs/error-codes)

Reply:
That page is for API error codes and this unknown error would include errors unrelated to API for example when you do workflow steps. So I don't think it is the right place for it.

Reply:
good point. I think there's talk of turning these into enums as a source of truth, so we can use this then.